pkg/subctl/cmd/gather/connectivity.go

```diff
@@ -0,0 +1,120 @@
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package gather
+
+import (
+	"bytes"
+	"io"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+
+	"github.com/submariner-io/submariner-operator/pkg/discovery/network"
+)
+
+const (
+	gatewayPods = "app=submariner-gateway"
+	routeagentPods = "app=submariner-routeagent"
+)
+
+func GetGatewayPodsLogs(clientSet kubernetes.Interface, clustername string) error {
+	pods, err := network.FindPods(clientSet, v1.ListOptions{LabelSelector: gatewayPods})
+
+	if err != nil || pods == nil {
+		return err
+	}
+	for _, pod := range pods.Items {
+		err := getPodLogs(clientSet, pod.Name, clustername)
+		if err != nil {
+			return errors.WithMessagef(err, "error getting logs for pod %q", pod.Name)
+		}
+	}
+	return nil
+}
+
+func GetRouteagentPodsLogs(clientSet kubernetes.Interface, clustername string) error {
+	pods, err := network.FindPods(clientSet, v1.ListOptions{LabelSelector: routeagentPods})
+
+	if err != nil || pods == nil {
+		return err
+	}
+	for _, pod := range pods.Items {
+		err := getPodLogs(clientSet, pod.Name, clustername)
+		if err != nil {
+			return errors.WithMessagef(err, "error getting logs for pod %q", pod.Name)
+		}
+	}
+	return nil
+}
+
+func getPodLogs(clientset kubernetes.Interface, podname string, clustername string) error {
+	logrequest := clientset.CoreV1().Pods("submariner-operator").GetLogs(podname, &corev1.PodLogOptions{})
+
+	logs, err := processLogStream(logrequest, podname)
+	if err != nil {
+		return err
+	}
+
+	err = writeToFile(podname, logs, clustername)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func processLogStream(logrequest *rest.Request, podname string) (string, error) {
+	logstream, err := logrequest.Stream()
+	if err != nil {
+		return "", errors.WithMessagef(err, "error opening log stream for pod %q", podname)
+	}
+	defer logstream.Close()
+
+	logs := new(bytes.Buffer)
+	_, err = io.Copy(logs, logstream)
+	if err != nil {
+		return "", errors.WithMessagef(err, "error converting log stream to string for pod %q", podname)
+	}
+	return logs.String(), nil
+}
+
+func writeToFile(podname, data, clustername string) error {
+	dirname := "submariner-" + time.Now().Format("20060102150405") //submariner-YYYYMMDDHHMMSS
+	if _, err := os.Stat(dirname); os.IsNotExist(err) {
+		err := os.MkdirAll(dirname, 0777)
+		if err != nil {
+			return errors.WithMessagef(err, "error creating directory %s", dirname)
+		}
+	}
+
+	filename := filepath.Join(dirname, clustername + "-" + podname + ".log")
+	f, err := os.OpenFile(filename, os.O_CREATE|os.O_APPEND|os.O_RDWR, 0777)
```

Comment:
```suggestion
	f, err := os.Create(filename)
```

Reply:
Create truncates the data while OpenFile will append the data to the existing file.

Reply:
right we shouldn't want to append although it shouldn't already exist unless we allowed the dir to be specified.  Also I think we want mode 0666 not 0777 as the latter gives read/write/execute access to everyone.